### ISSUE TITLE
After upgrading to 1.0.31, console.log no prints

### ISSUE THREAD

**Issue by akirarika** [oven-sh/bun#9428]
### What version of Bun is running?

1.0.31

### What platform is your computer?

Darwin 23.4.0 arm64 arm

### What steps can reproduce the bug?

I am developing a framework based on Bun, but after upgrading to 1.0.31, I found that although the program is functioning normally, some console.log outputs are no longer printed to the console. I have tried to locate the issue, but haven't found it yet, so I don't have a minimal reproducible code snippet at the moment. However, I will continue to try my best to find it later.

First, clone this repository: https://github.com/southern-aurora/loongbao

Install dependencies: `bun i`

Install version 1.0.30 of bun:

```sh
curl -fsSL https://bun.sh/install | bash -s "bun-v1.0.30"
```

Run the following command:

```sh
bun run ./node_modules/loongbao/c.ts test "/a-sandbox/hello-world"
```

Output:

```
Hello World
🧊 Loongbao is running on : /Users/i/homespace/loongbao
RUN_MODE API_TEST
🧊 Loongbao Api Testing..

✅ DIRECT --  Basic | Path: src/apps/a-sandbox/hello-world.ts | Case: 1 | Time: 16ms

✅ All tests passed.
🧊 Loongbao Api Testing took 0.02s
```

Once I upgraded to 1.0.31, although the program actually runs correctly, the output is missing.

```sh
curl -fsSL https://bun.sh/install | bash -s "bun-v1.0.31"
```

Run the following command:

```sh
bun run ./node_modules/loongbao/c.ts test "/a-sandbox/hello-world"
```

Output:

```
Hello World
```

The output afterwards is all gone.

Although there is no output, it seems that the code is actually executing correctly.

**Comment by Jarred-Sumner** [comment]
I am able to reproduce 

my guess is it’s a shell bug but I’m not sure yet

**Comment by alexanderclarktx** [comment]
maybe unrelated, but fwiw, my builds broke with newest bun docker image

oven/bun:1.0-slim ❌
```
$ bun --cwd server start
$ bun src/PiggoApi.ts
error: Module not found "src/PiggoApi.ts"
```

oven/bun:1.0.30-slim ✅
```sh
$ bun --cwd server start
$ bun src/PiggoApi.ts
包 http://0.0.0.0:6332/
```

**Comment by akirarika** [comment]
I'm not sure if this is the problem, but I just changed the code of $ shell to execute `"bash -c "scripts"` using `Bun.spawn`, which restored its normal output.

```ts
// await $`${{ raw: packageJson.scripts.start }}`.env({
//   ...
// });

// Now it is working:
await exec(rootPath, ["bash", "-c", packageJson.scripts.start], {
  env: {
    ...
  }
});
```

exec is just a simple wrapper for Bun.spawn, code:

```ts
export const exec = async (cwd: string, command: Array<string>, options: Partial<SpawnOptions.OptionsObject> = {}) => {
  return new Promise((resolve, reject) => {
    if (!("cwd" in options)) options.cwd = cwd;
    if (!("stdin" in options)) options.stdin = "inherit";
    if (!("stdout" in options)) options.stdout = "inherit";
    if (!("env" in options)) options.env = { ...env };

    options.onExit = (proc, exitCode, signalCode, error) => {
      // eslint-disable-next-line prefer-promise-reject-errors
      if (exitCode !== 0) reject({ proc, exitCode, signalCode, error });
      else resolve({ proc, exitCode, signalCode, error });
    };

    try {
      Bun.spawn(command, options);
    } catch (error) {
      reject(error);
    }
  });
};
```

**Comment by akirarika** [comment]
I really like Bun and $ Shell, they are awesome. Now when I write small utility scripts, I don't want to use Python or Shellscript anymore. Bun is changing the world

**Comment by reaby** [comment]
Okay... i have the same bug, more to this insight: 
If i run the app in wsl console normally all works as it should... but if i run it in vscode with debug and debug console it stops printing the log lines.

**Comment by zackradisic** [comment]
This seems to be fixed with #9402
